kubernetes-deployment/src/chart/templates/_helpers.tpl

```diff
@@ -18,16 +18,16 @@ Common labels
 */}}
 {{- define "application.labels" -}}
 {{ include "application.selectorLabels" . }}
-{{- with .Values.labels }}
+{{- with .Values.commonLabels }}
 {{ toYaml . }}
 {{- end }}
-app.kubernetes.io/name: {{ include "application.fullname" . }}
+app.kubernetes.io/chart: {{ include "application.chart" . }}
```

Comment:
Labels end up being this:
```
Labels:                 app.kubernetes.io/chart=foo-0.0.6
                        app.kubernetes.io/managed-by=Helm
                        app.kubernetes.io/name=chris-local-test-0000
```

We end up using the `application.fullname` for the selector label, which is the `md_metadata.name_prefix` which should guarantee uniqueness for the selectors.

Reply:
^ this seems fine